tests/integration/targets/module_vault_kv2_delete/tasks/module_vault_kv2_delete_test.yml

```
@@ -0,0 +1,220 @@
+              - "kv2_result['result']['data']['versions']['5']['deletion_time'] is search(regex_secret_version_not_deleted)"
+            fail_msg: 'Test Seed value did not contain expected data.'
+
+        - name: Try kv2 delete latest version in check mode
+          register: kv2_result
+          community.hashi_vault.vault_kv2_delete:
+            path: "{{ vault_kv2_versioned_path }}/secret6"
+          check_mode: true
+
+        - name: Check secret versions
+          register: kv2_result
+          vault_ci_kv2_metadata_read:
+            path: "{{ vault_kv2_versioned_path }}/secret6"
+            mount_point: '{{ vault_kv2_mount_point }}'
+
+        - assert:
+            that:
+              - *assert_basics
+              - "kv2_result['result']['data']['versions']['1']['deletion_time'] is search(regex_secret_version_not_deleted)"
+              - "kv2_result['result']['data']['versions']['2']['deletion_time'] is search(regex_secret_version_not_deleted)"
+              - "kv2_result['result']['data']['versions']['3']['deletion_time'] is search(regex_secret_version_not_deleted)"
+              - "kv2_result['result']['data']['versions']['4']['deletion_time'] is search(regex_secret_version_not_deleted)"
+              - "kv2_result['result']['data']['versions']['5']['deletion_time'] is search(regex_secret_version_not_deleted)"
+            fail_msg: 'Secret version was deleted while in check mode.'
+
+
+        - name: Try kv2 delete specific version in check mode
+          register: kv2_result
+          community.hashi_vault.vault_kv2_delete:
+            path: "{{ vault_kv2_versioned_path }}/secret6"
+            versions: [1, 3]
+          check_mode: true
+
+        - name: Check secret versions
+          register: kv2_result
+          vault_ci_kv2_metadata_read:
+            path: "{{ vault_kv2_versioned_path }}/secret6"
+            mount_point: '{{ vault_kv2_mount_point }}'
+
+        - assert:
+            that:
+              - *assert_basics
+              - "kv2_result['result']['data']['versions']['1']['deletion_time'] is search(regex_secret_version_not_deleted)"
+              - "kv2_result['result']['data']['versions']['2']['deletion_time'] is search(regex_secret_version_not_deleted)"
+              - "kv2_result['result']['data']['versions']['3']['deletion_time'] is search(regex_secret_version_not_deleted)"
+              - "kv2_result['result']['data']['versions']['4']['deletion_time'] is search(regex_secret_version_not_deleted)"
+              - "kv2_result['result']['data']['versions']['5']['deletion_time'] is search(regex_secret_version_not_deleted)"
+            fail_msg: 'Secret version was deleted while in check mode.'
+
+        - name: Try kv2 delete version 1 and 3
+          register: kv2_result
+          community.hashi_vault.vault_kv2_delete:
+            path: "{{ vault_kv2_versioned_path }}/secret6"
+            versions:
+              - 1
+              - 3
+        - assert:
```

spacing is a nit

But something is strange here. I don't understand how this assertion is passing. `kv2_result` should contain nothing but `changed=true` and `data={}`, so how are the basics assertions not failing?

```suggestion

        - assert:
```

---

Ah! I have figured this out. We cannot use YAML anchors like this.

With the following:
```yaml
- assert:
    that: &a
      - 1
      - 2
```

The anchor is referring to the entire list value: `[1, 2]`.

When using it like this:
```yaml
- assert:
    that:
      - *a
      - b
```

Then, what we have for the list of assertions then is actually this: `[[1,2], "b"]`

So the first list item is a non-empty list which always evaluates to true.

For dictionaries, YAML anchors have the merge operator `<<` but there is no such thing for list.

We don't have a lot of good options for this. Either duplicate all of the "basic" assertions, or use vars in a way like this:

```
- name: some block
  vars:
    basic_assertions: &basics
      - 1
      - 2
 
- name: simple case - basics only - option 1
  assert:
    that: *basics

- name: simple case - basics only - option 2
  assert:
    that: '{{ basic_assertions }}'

- name: complex 1 - literals
  assert:
    that: "{{ basic_assertions + ['condition 1', 'condition 2'] }}"

- name: complex 2 - vars
  vars:
    these_assertions:
      - condition 1
      - condition 2
  assert:
    that: '{{ basic_assertions + these_assertions }}'
```

---

I feel its a lot more readable to just duplicate the lines from the anchor where its needed (which is only a handful).. Cleaning up some of that now and will push something up once i run some local tests